config-model/src/main/java/com/yahoo/vespa/model/application/validation/change/RestartOnDeployForOnnxModelChangesValidator.java

```diff
@@ -104,10 +105,13 @@ private static void setRestartOnDeployAndAddRestartAction(List<ConfigChangeActio
     private static boolean enoughMemoryToAvoidRestart(ApplicationContainerCluster clusterInCurrentModel,
                                                       ApplicationContainerCluster cluster,
                                                       DeployLogger deployLogger) {
+        List<ApplicationContainer> containers = cluster.getContainers();
+        if (containers.isEmpty()) return true;
+
         double currentModelCostInGb = onnxModelCostInGb(clusterInCurrentModel);
         double nextModelCostInGb = onnxModelCostInGb(cluster);
 
-        double totalMemory = cluster.getContainers().get(0).getHostResource().realResources().memoryGb();
+        double totalMemory = containers.get(0).getHostResource().realResources().memoryGb();
```

Comment:
You may want to do the calculation similar to how it's done in `ApplicationContainerCluster` - use the node with lowest memory.
```java
getContainers().stream().mapToDouble(c -> c.getHostResource().realResources().memoryGb()).min().orElseThrow()
```

Reply:
Made a separate PR for this: https://github.com/vespa-engine/vespa/pull/29958